core/src/main/java/io/micronaut/core/type/TypeInformation.java

```diff
@@ -284,11 +310,193 @@ default boolean isArray() {
+                case QUALIFIED -> "@" + annotationRef + members;
+                case SHORTENED -> "@" + NameUtils.getShortenedName(annotationRef) + members;
+                case ANSI_SIMPLE ->
+                    AnsiColour.yellow("@" + NameUtils.getSimpleName(annotationRef)) + members;
+                case ANSI_QUALIFIED -> AnsiColour.yellow("@" + annotationRef) + members;
+                case ANSI_SHORTENED ->
+                    AnsiColour.yellow("@" + NameUtils.getShortenedName(annotationRef)) + members;
+            };
+        }
+
+        /**
+         * Get a type string for the given format.
+         *
+         * @param format   The format
+         * @param type     The type
+         * @param generics The generics
+         * @return the type string
+         */
+        public static @NonNull String getTypeString(
+            @NonNull TypeFormat format,
+            @NonNull Class<?> type,
+            @NonNull Map<String, Argument<?>> generics) {
+            String typeName = switch (format) {
+                case SIMPLE -> type.getSimpleName();
+                case QUALIFIED -> type.getCanonicalName();
+                case SHORTENED -> NameUtils.getShortenedName(type.getTypeName());
+                case ANSI_SIMPLE -> AnsiColour.cyan(type.getSimpleName());
+                case ANSI_QUALIFIED -> AnsiColour.cyan(type.getCanonicalName());
+                case ANSI_SHORTENED ->
+                    AnsiColour.cyan(NameUtils.getShortenedName(type.getCanonicalName()));
+            };
+            StringBuilder returnType = new StringBuilder(typeName);
+            if (!generics.isEmpty()) {
+                returnType
+                    .append(format.isAnsi() ? AnsiColour.brightCyan("<") : "<")
+                    .append(generics.values()
+                        .stream()
+                        .map(arg -> arg.getTypeString(format))
+                        .collect(Collectors.joining(", ")))
+                    .append(format.isAnsi() ? AnsiColour.brightCyan(">") : ">");
+            }
+            return returnType.toString();
+        }
+
+        /**
+         * Get a type string for the given format.
+         *
+         * @param format             The format
+         * @param type               The type
+         * @param generics           The generics
+         * @param annotationMetadata The annotation metadata
+         * @return the type string
+         */
+        public static @NonNull String getBeanTypeString(
+            @NonNull TypeFormat format,
+            @NonNull Class<?> type,
+            @NonNull Map<String, Argument<?>> generics,
+            @NonNull AnnotationMetadata annotationMetadata) {
+            String typeFormat = TypeFormat.getTypeString(
+                format,
+                type,
+                generics
+            );
+            Optional<String> q = annotationMetadata.getAnnotationNameByStereotype(AnnotationUtil.QUALIFIER)
```

Comment:
Maybe you can use similar code to `Qualifiers.forArgument`

Reply:
code here is in core which can't reference inject